### ISSUE TITLE
Add command for immediate value changes to the TextBox control

### ISSUE THREAD

**Issue by ENgateman** [riganti/dotvvm#425]
Add a command that would be called from within an onkeypress event handler with a certain timeout so that the postback gets sent only after the user has stopped typing.

It would be very useful for creating search boxes.

We should also consider creating a separate control (maybe SearchTextBox?) that would contain this command so that we don't pollute the TextBox control with too many properties.

**Comment by djanosik** [comment]
Not sure whether to extend the TextBox or create another control. But the command name should not be related to events used on client-side. We need to use `input` event in modern browser and a combination of other events in old browsers.

**Comment by exyi** [comment]
I'd say, that `OnInput` name looks good, and we can probably safely use the `input` event, see https://caniuse.com/#search=input it works everywhere, except Opera Mini.

The other problem is how to the postback throttling. In general, it would be nice to have a postback handler that would allow you to throttle any command, but it would be too XAMLy if you'd have to specify the handler, so we could make an attached property group, that would alias that to something like `Postback.Throttle-OnInput="700"`.

**Comment by djanosik** [comment]
`OnInput` conflicts with client-side event attribute and breaks our naming conventions. How about `Changing`?

**Comment by tomasherceg** [comment]
What about TextInput?

S pozdravem
Tomáš Herceg

RIGANTI s.r.o. | DOTNETCOLLEGE s.r.o. | DotVVM
www.riganti.cz | www.dotnetcollege.cz | www.dotvvm.com
________________________________

**Comment by ENgateman** [comment]
So to sum this up:

- We agreed we'd use the input event, therefore using the word 'key' in the property name would be misleading.
- OnInput looks good, but it breaks our naming convention which is to use the past form of a verb that describes why the command has been invoked (e.g. Changed, UploadCompleted, ItemDropped)
- Changing breaks the convention too, as 'changing' is a present participle
- TextInput doesn't break the convention, but only because input already is in its past form. It can be still quite misleading I think.

Updated list of options:

- ~OnKeyPressChanged~
- ~OnInput~
- ~Changing~
- TextInput
- TextInputted - alternative past participle of 'input"
- Inputted - we can also omit 'text' completely

We could also add a ChangedMode property that would control whether to use the 'change' event or the 'input' event. What do you think?

**Comment by djanosik** [comment]
We use `UpdateTextOnInput` property in BP AutoComplete control. How about to add another boolean property `PostBackOnInput`? And use the original `Changed` command.

**Comment by tomasherceg** [comment]
I like `TextInput`.

**Comment by djanosik** [comment]
Then I would use `Input` because we have `Changed` and not `TextChanged`.

**Comment by tomasherceg** [comment]
I am not sure whether `Input` is intuitive enough.

**Comment by ENgateman** [comment]
I don't find `Input` any more intuitive than `TextInput`. Both share the problem of being a noun, a present participle and a past participle at the same time and that makes it difficult to tell whether they are supposed to be services, methods or commands, at least at first glance anyway.

**Comment by tomasherceg** [comment]
We have added `UpdateTextOnInput` recently, which triggers the `Changed` event immediately.

The issue also mentions throttling that could be done using a postback handler (although it is not easy). Closing, and I'll open a separate issue for throttling postback handler.
